azurerm/internal/services/network/resource_arm_virtual_hub_connection.go

```diff
@@ -0,0 +1,301 @@
+	defer locks.UnlockByName(id.Name, virtualHubResourceName)
+
+	virtualHub, err := client.Get(ctx, resourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(virtualHub.Response) {
+			return fmt.Errorf("Virtual Hub %q was not found in Resource Group %q", id.Name, resourceGroup)
+		}
+
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): %+v", id.Name, resourceGroup, err)
+	}
+
+	if virtualHub.VirtualHubProperties == nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): `properties` was nil", id.Name, resourceGroup)
+	}
+
+	props := *virtualHub.VirtualHubProperties
+
+	var connections []network.HubVirtualNetworkConnection
+	if props.VirtualNetworkConnections != nil {
+		connections = *props.VirtualNetworkConnections
+	}
+
+	name := d.Get("name").(string)
+
+	if features.ShouldResourcesBeImported() {
+		if connection, _ := findVirtualHubConnection(name, connections); connection != nil {
+			return tf.ImportAsExistsError("azurerm_virtual_hub_connection", *connection.ID)
+		}
+	}
+
+	connection := network.HubVirtualNetworkConnection{
+		Name: utils.String(name),
+		HubVirtualNetworkConnectionProperties: &network.HubVirtualNetworkConnectionProperties{
+			RemoteVirtualNetwork: &network.SubResource{
+				ID: utils.String(d.Get("remote_virtual_network_id").(string)),
+			},
+			AllowHubToRemoteVnetTransit:         utils.Bool(d.Get("hub_to_vitual_network_traffic_allowed").(bool)),
+			AllowRemoteVnetToUseHubVnetGateways: utils.Bool(d.Get("vitual_network_to_hub_gateways_traffic_allowed").(bool)),
+			EnableInternetSecurity:              utils.Bool(d.Get("internet_security_enabled").(bool)),
+		},
+	}
+	connections = append(connections, connection)
+	virtualHub.VirtualHubProperties.VirtualNetworkConnections = &connections
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, id.Name, virtualHub)
+	if err != nil {
+		return fmt.Errorf("Error adding Connection %q to Virtual Hub %q (Resource Group %q): %+v", name, id.Name, resourceGroup, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for addition of Connection %q to Virtual Hub %q (Resource Group %q): %+v", name, id.Name, resourceGroup, err)
+	}
+
+	virtualHub, err = client.Get(ctx, resourceGroup, id.Name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): %+v", id.Name, resourceGroup, err)
+	}
+	if virtualHub.VirtualHubProperties == nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): `properties` was nil", id.Name, resourceGroup)
+	}
+	if virtualHub.VirtualHubProperties.VirtualNetworkConnections == nil {
+		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): `properties.VirtualNetworkConnections` was nil", id.Name, resourceGroup)
+	}
+
+	connections = *virtualHub.VirtualHubProperties.VirtualNetworkConnections
+	newConnection, _ := findVirtualHubConnection(name, connections)
```

Comment:
I think if you moved the checks into the find function and took the vhub this would be much more clear
```suggestion
	newConnection, _ := findVirtualHubConnection(name, virtualHub)
```

Reply:
Updated